packages/js/src/integrations-page/site-kit-integration.js

```diff
@@ -50,20 +58,63 @@ const fetchJson = async( options ) => {
 	}
 };
 
+/**
+ * The wrapper for the top section of the footer.
+ *
+ * @param {JSX.Element} children The children.
+ * @param {string} className The class name.
+ * @returns {JSX.Element} The top footer wrapper.
+ */
+const TopFooterWrapper = ( { children, className = "" } ) => {
+	return <span className={ classNames( "yst-flex yst-justify-between yst-pb-4 yst-border-b yst-mb-6 yst-border-slate-200 yst--mt-2", className ) }>
+		{ children }
+	</span>;
+};
+
+TopFooterWrapper.propTypes = {
+	children: PropTypes.node.isRequired,
+	className: PropTypes.string,
+};
+
 /**
  * The successfully connected component.
  *
  * @returns {JSX.Element} The SuccessfullyConnected component.
  */
 const SuccessfullyConnected = () => {
-	return <span className="yst-flex yst-justify-between yst-pb-4 yst-border-b yst-mb-6 yst-border-slate-200 yst--mt-2">
-		<span className="yst-text-slate-700 yst-font-medium">
-			{ __( "Successfully connected", "wordpress-seo" ) }
-		</span>
+	return <TopFooterWrapper className="yst-text-slate-700 yst-font-medium">
+		{ __( "Successfully connected", "wordpress-seo" ) }
 		<CheckIcon
 			className="yst-h-5 yst-w-5 yst-text-green-400 yst-flex-shrink-0"
 		/>
-	</span>;
+	</TopFooterWrapper>;
+};
+
+/**
+ * The no permission warning component.
+ *
+ * @param {Capabilities} capabilities The capabilities.
+ * @param {number} currentStep The current step.
+ *
+ * @returns {JSX.Element} The no permission warning component.
+ */
+const NoPermissionWarning = ( { capabilities, currentStep } ) => {
+	if ( ! capabilities.installPlugins && currentStep < 3 && currentStep !== -1 ) {
```

Comment:
Perhaps you can add the `const successfullyConnected = currentStep === -1;` line above here? That way we have some documention in code about the currentStep being or not being -1?

Or just add a comment about it. And while you're at it, what step < 3 is, and below step > 2?